Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -255,19 +254,18 @@ namespace Babylon::Polyfills::Internal
             }
         }
 
-        m_request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [env{info.Env()}, this](arcana::expected<void, std::exception_ptr> result) {
-            if (result.has_error())
-            {
-                Napi::Error::New(env, result.error()).ThrowAsJavaScriptException();
-                return;
-            }
-
+        m_request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [this]() {
             SetReadyState(ReadyState::Done);
             RaiseEvent(EventType::LoadEnd);
 
             // Assume the XMLHttpRequest will only be used for a single request and clear the event handlers.
             // Single use seems to be the standard pattern, and we need to release our strong refs to event handlers.
             m_eventHandlerRefs.clear();
+        }).then(arcana::inline_scheduler, arcana::cancellation::none(), [env = info.Env()](arcana::expected<void, std::exception_ptr> result) {
+            if (result.has_error())
+            {
+                Napi::Error::New(env, result.error()).ThrowAsJavaScriptException();
```

Comment:
I know you are only moving this code in this PR (not introducing it), but can you remind me why `ThrowAsJavaScriptException` is being used in this case? Since contractually XMLHttpRequest already has a way of communicating errors (the `error` event), I would have thought we'd see that being used here. I thought `ThrowAsJavaScriptException` was only used if you aren't using C++ exceptions at all or if there is effectively a fire and forget async call from JS to native (so there is no other way to communicate the error back to JS). Am I remembering that wrong? I also don't remember when on the JS side this thrown error will be observed. 🤔

Reply:
Hmm, I don't know. Ideally, we should match the browser's behavior.